Snakefile

```diff
@@ -0,0 +1,94 @@
+import pandas as pd
+import sourmash
+import os
+
+# create wildcard variables for workflow
+metadata = pd.read_csv("inputs/metadata-paired-mgx-mtx.tsv", sep = "\t") # read in metadata as a pandas dataframe
+MTX = metadata['mtx_run_accession'].unique().tolist()                    # make run accession in mtx col into list
+MGX = metadata['mgx_run_accession'].unique().tolist()                    # make run accession in mgx col into list
+RUN_ACCESSIONS = MGX + MTX                                               # combine mtx and mgx into one list with all run accessions
+SAMPLES = metadata['sample_name'].unique().tolist()                      # make a list of sample names
+KSIZES = [21, 31, 51]                                                    # create a list of k-mer sizes for the workflow
+
+rule all:
+    input: expand("outputs/sourmash_sketch_subtract/{sample}_k{ksize}.sig", sample = SAMPLES, ksize = KSIZES)
+
+rule sourmash_sketch:
+    """
+    Use fastq-dump to download sequencing data for a given accession.
+    Pipe the sequencing data directly to a sketch without writing to disk.
+    Using fastq-dump like this dumps both single end and paired end reads into std out, which can then be piped into sourmash sketch
+    """
+    output: "outputs/sourmash_sketch/{run_accession}.sig"
+    conda: 'envs/sourmash.yml'
+    shell:'''
+    fastq-dump --disable-multithreading --fasta 0 --skip-technical --readids --read-filter pass --dumpbase --split-spot --clip -Z {wildcards.run_accession} | 
+        sourmash sketch dna -p k=21,k=31,k=51,scaled=200,abund --name {wildcards.run_accession} -o {output} -
```

Comment:
extra `- `after `-o {output} -` ?

Reply:
Ah that's supposed to be there, but good eagle eye! that means, from the stdout from our pipe `|`, put it here in the command.